### PR TITLE
K8s: Add client

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

* Adds K8s client based on bridge client in #60207. The main change is a new function called `GetResourceClient` that uses a [DynamicClient](https://pkg.go.dev/k8s.io/client-go/dynamic#DynamicClient) and [DeferredDiscoveryRESTMapper](https://pkg.go.dev/k8s.io/client-go/restmapper#DeferredDiscoveryRESTMapper) to get a [Resource](https://pkg.go.dev/k8s.io/client-go/dynamic#ResourceInterface) client that can be used for CRUD operations.
* Removes client from `k8saccess`

**Why do we need this feature?**

**Who is this feature for?**

@grafana/multitenancy-squad 

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/k8s/client/clientset.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	ErrCRDAlreadyRegistered = errors.New("error registering duplicate CRD")
+)
+
+// Clientset is the clientset for Kubernetes APIs.
+// It provides functionality to talk to the APIs as well as register new API clients for CRDs.
+type Clientset struct {
+	config *rest.Config
+
+	clientset kubernetes.Interface
+	extset    apiextensionsclient.Interface
```
Comment (toddtreece) [inline pkg/services/k8s/client/clientset.go:33]: i switched these to interfaces instead of pointers so that it might be easier to write tests

**Review by ryantxu — APPROVED** [review]
added codeowners

**Comment by ryantxu** [comment]
I like this -- but we should remove https://github.com/grafana/grafana/blob/main/pkg/services/store/k8saccess/service.go#L31 as part of this PR then

File: pkg/services/dashboards/service/service.go [hunk 2]
```diff
@@ -10,13 +10,9 @@ import (
 func ProvideSimpleDashboardService(
 	features featuremgmt.FeatureToggles,
 	svc *DashboardServiceImpl,
-	k8s k8saccess.K8SAccess,
```
Comment (ryantxu) [inline pkg/services/dashboards/service/service.go:13]: should this be the new client?

Comment (toddtreece) [reply pkg/services/dashboards/service/service.go:13]: this is restructured in the next branch, so i think it will be easier to handle it there

**Review by ryantxu — APPROVED** [review]

**Review by ryantxu — APPROVED** [review]
LGTM

**Pull request merged**